### ISSUE TITLE
open URLs with chrome or safari or firefox as an option over Lynx

### ISSUE THREAD

**Issue by elliottw** [xolox/vim-shell#18]

**Comment by xolox** [comment]
- On what platform? And which version?
- Which web browsers do you have installed?
- Can you define exactly what you do, what happens now, and what should happen instead according to you?

**Comment by elliottw** [comment]
Sorry about that:
Mac 10.8.4
lynx, chrome, safari, firefox
right now links are opened in lynx. i would like to set that behavior to open lynx in chrome.

thank you

**Comment by xolox** [comment]
I suppose you're running Vim in the console?

**Comment by xolox** [comment]
I use Mac OS X myself, but always using MacVim, never console Vim. I guess I never gave this any thought; on Mac OS X I don't even think it's possible not to have a GUI available, so really it never makes sense to resort to a command line web browser; who wouldn't have a proper GUI web browsers installed nowadays? :-)

**Comment by elliottw** [comment]
I guess it works as expected in macvim. I switch between iTerm2 and MacVim depending on what I'm doing. If I'm navigating directories in the console and pop open vim, for some reason I expect the same behavior. Feel free to call this a feature and not a bug.

**Comment by xolox** [comment]
Thanks for reporting this issue, I think it's now fixed... The relevant change is not in the [vim-shell plug-in](https://github.com/xolox/vim-shell) but in my [vim-misc plug-in](https://github.com/xolox/vim-misc), so make sure you install the latest version (1.6.2) of vim-misc. Can you confirm here whether this fixes the issue for you as well? Thanks in advance.

**My analysis of the situation (based on my Mac OS X 10.7.5 installation):**

I believe you are using `/usr/bin/vim` instead of `/Applications/MacVim.app/Contents/MacOS/Vim`. I only recently noticed myself that when you run `vim` in the console on Mac OS X, you get some crappy Vim variant. I say crappy because that version of Vim has some strange defaults that break several of my plug-ins, i.e. I've had to work around `/usr/bin/vim` on Mac OS X before :-).

However only after you reported this issue, I noticed that `/usr/bin/vim` on Mac OS X does not realize it is running on a Mac. Vim has functions to report the platform, but in this version of Vim they don't work! I think this is the reason why you saw the issue you reported here.

I've now implemented a workaround that should properly detect Mac OS X, no matter which version of Vim is used. In case anyone is curious: I run `uname` and check if it outputs `Darwin`.

**Comment by elliottw** [comment]
ran git pull origin master from .vim/bundle/vim-misc, still opening links in lynx.
ran mvim -v in terminal and then :Open still opening in lynx.
uname response is darwin

I don't know what the deal is.

**Comment by xolox** [comment]
Thanks for the quick response. Can you try something for me? I added a bunch of debugging messages in vim-misc version 1.6.2 which might come in handy now :-). To make them visible you can execute the following Vim command:

``` vim
:verbose Open http://www.vim.org/
```

In verbose mode the messages will be added to Vim's message buffer and you can recall previous messages by running the Vim command `:messages`. Would you mind reporting the messages here?

**Comment by elliottw** [comment]
with vanilla vim:

> Messages maintainer: Bram Moolenaar Bram@vim.org
> shell.vim 0.13: Opening URL or e-mail address 'http://www.vim.org/'.
> vim-misc 1.6.2: Using command line web browser because no GUI seems to be available ..
> vim-misc 1.6.2: Checking whether 'lynx' command line web browser is installed ..
> vim-misc 1.6.2: Found 'lynx', using it to open 'http://www.vim.org/' ..
> Press ENTER or type command to continue

with mvim -v

> Messages maintainer: Bram Moolenaar Bram@vim.org
> shell.vim 0.13: Opening URL or e-mail address 'http://www.vim.org/'.
> vim-misc 1.6.2: Using command line web browser because no GUI seems to be available ..
> vim-misc 1.6.2: Checking whether 'lynx' command line web browser is installed ..
> vim-misc 1.6.2: Found 'lynx', using it to open 'http://www.vim.org/' ..
> Press ENTER or type command to continue

**Comment by xolox** [comment]
Hmm, strange, apparently the detection of Mac OS X is still somewhat broken, at least for your version of Mac OS X (10.8.4). It works for me on 10.7.5 and 10.8.0 (I just tested again to be absolutely sure :-). Can you run the following Vim command and report the literal output here?

``` vim
:echo xolox#misc#os#exec({'command': 'uname', 'check': 0})
```

What I'm trying to find out is the exact output of the `uname` command (including whitespace, empty lines, etc.), because I think that's where it's going wrong.

**Comment by elliottw** [comment]
from vim:

```
{'stderr': [], 'exit_code': 0, 'stdout': ['Darwin'], 'command': 'sh -c ''(uname) 1>''\''''/var/folders/_0/m23ptrtn4z927gq285qv08bc0000g
p/T/vfrGfri/0''\'''' 2>''\''''/var/folders/_0/m23ptrtn4z927gq285qv08bc0000gp/T/vfrGfri/1''\'''''''}
Press ENTER or type command to continue
```

**Comment by xolox** [comment]
Sorry for all the back and forth and thanks for all of the information you've provided! I just published vim-misc version 1.6.3 which I think should finally fix the issue you reported... Can you confirm?

**Comment by elliottw** [comment]
Success! Now when I use vim in the terminal, it opens in my system wide default browser (chrome).
Thanks!
